### ISSUE TITLE
Allow to filter each generator by a selector like a labelSelector

### ISSUE THREAD

**Issue by farodin91** [argoproj/applicationset#361]
We are currently use applicationset to deploy multiple applications into a same set of clusters, but some application are just need in certain clusters. It would be greated to use a selector to post filter any generator output.

Idea:
```yaml
apiVersion: argoproj.io/v1alpha1
kind: ApplicationSet
metadata:
  name: guestbook
spec:
  generators:
  - list:
      elements:
        - cluster: engineering-dev
          url: https://kubernetes.default.svc
          env: staging
        - cluster: engineering-prod
          url: https://kubernetes.default.svc
          env: prod
    selector:
      matchLabels:
        env: staging
  template:
    metadata:
      name: '{{cluster}}-guestbook'
    spec:
      project: default
      source:
        repoURL: https://github.com/argoproj-labs/applicationset.git
        targetRevision: HEAD
        path: examples/list-generator/guestbook/{{cluster}}
      destination:
        server: '{{url}}'
        namespace: guestbook
```

**Comment by wtam2018** [comment]
@farodin91 can you please update your proposal with labelSelector?

**Comment by farodin91** [comment]
@wtam2018 I updated the doc.

**Comment by adabuleanu** [comment]
Is this feature available from 2.3 release? Although the docs say 2.3, I could only find it in 2.5 release.
